src/graphforge/datasets/loaders/compression.py

```
@@ -90,25 +140,40 @@ def extract_tar_zst(archive_path: Path, extract_to: Path) -> None:
-
-        # Decompress and extract tar
-        with (
-            dctx.stream_reader(compressed_file) as reader,
-            tarfile.open(fileobj=reader, mode="r|") as tar,
-        ):
-            # Extract with path validation
+    # Open zstd-compressed file and decompress to temporary tar
+    import tempfile
+
+    # Use delete=False for Windows compatibility (avoids exclusive lock issues)
+    temp_tar_file = tempfile.NamedTemporaryFile(suffix=".tar", delete=False)  # noqa: SIM115
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (non-enabled: `SIM115`)

Remove unused `noqa` directive

(RUF100)